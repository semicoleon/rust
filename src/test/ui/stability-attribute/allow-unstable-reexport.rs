// Allow an unstable re-export without requiring a feature gate. https://github.com/rust-lang/rust/issues/94972

// aux-build:lint-stability.rs
// aux-build:lint-stability-reexport.rs
#![feature(staged_api)]
#![stable(feature = "lint_stability", since = "1.0.0")]

extern crate lint_stability;
extern crate lint_stability_reexport;

use lint_stability::unstable;
// We want to confirm that using a re-export through another crate behaves the same way as using an item directly
use lint_stability_reexport::unstable_text;

fn main() {
    // Since we didn't enable the feature in this crate, we still can't use these items, even though they're in scope from the `use`s which are now allowed.
    unstable(); //~ ERROR use of unstable library feature 'unstable_test_feature'
    unstable_text(); //~ ERROR use of unstable library feature 'unstable_test_feature'
}