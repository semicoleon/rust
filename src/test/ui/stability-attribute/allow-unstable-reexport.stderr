error[E0658]: use of unstable library feature 'unstable_test_feature'
  --> $DIR/allow-unstable-reexport.rs:20:9
   |
LL | pub use lint_stability::unstable as unstable2;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add `#![feature(unstable_test_feature)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'unstable_test_feature'
  --> $DIR/allow-unstable-reexport.rs:26:5
   |
LL |     unstable();
   |     ^^^^^^^^
   |
   = help: add `#![feature(unstable_test_feature)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'unstable_test_feature': text
  --> $DIR/allow-unstable-reexport.rs:27:5
   |
LL |     unstable_text();
   |     ^^^^^^^^^^^^^
   |
   = help: add `#![feature(unstable_test_feature)]` to the crate attributes to enable

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0658`.
